### DISCUSSION TITLE
[NOT A CRUCIAL QUESTION] Are built-in torch functions really "faster"?

### DISCUSSION THREAD

**Discussion by sachinlodhi** [mrdbourke/pytorch-deep-learning#1099] [Q&A]
So I was watching the matrix multiplication section and was trying to compare the manual (for-loop) method and the PyTorch built-in method i.e. torch.matmul().
Interestingly, I found that, even for 1D vector, for loop was faster than torch.matmul().
The screenshot is attached. 
![Screenshot from 2024-09-23 11-30-09](https://github.com/user-attachments/assets/e145854b-b4fd-4bac-9e42-f2ee23b6ff3b)

**Comment by LuluW8071** [comment]
Interesting experiment. Yes they could be faster for 1D ones. But for 2D and higher dimensions, torch is faster. This is because `matmul` and other operations are written in C++ which is faster than regular python 

[Matmul refrence docs](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/LinearAlgebra.cpp#L1450)
